pkg/apis/application/v1alpha1/types.go

```diff
@@ -2891,6 +2891,12 @@ func SetK8SConfigDefaults(config *rest.Config) error {
 	config.Timeout = K8sServerSideTimeout
 
 	config.Transport = tr
+	maxRetries := env.ParseInt64FromEnv(utilhttp.EnvRetryMax, 0, 1, math.MaxInt64)
+	if maxRetries > 0 {
+		backoffDurationMS := env.ParseInt64FromEnv(utilhttp.EnvRetryBaseBackoff, 100, 1, math.MaxInt64)
+		backoffDuration := time.Duration(backoffDurationMS) * time.Millisecond
+		config.WrapTransport = utilhttp.WithRetry(maxRetries, backoffDuration)
+	}
```

Comment:
For better discovery, we should add these env vars to the cli [here](https://github.com/argoproj/argo-cd/blob/c47ea326cb5e731e30b2e1ec9b41c38fe53b275a/cmd/argocd-application-controller/commands/argocd_application_controller.go#L190) and pass them down. Also might be a good idea to document these in env in the docs.